### PR TITLE
MB-14433 React Admin Upgrade

### PR DESCRIPTION
**Pull request by PDickeyTruss**
## Summary

The focus of this PR is the upgrade of react-admin to v4. However, due to the interdependence between the two, this PR also includes the react-router upgrade work. All react-routing upgrade related work was previously code reviewed and user tested in PR https://github.com/transcom/mymove/pull/10020. The code changes needed to upgrade react-admin are detailed in the [upgrade guide](https://marmelab.com/react-admin/Upgrade.html). 

The admin app should have all the same functionality it did before the upgrade and only differ slightly on styling as the new react-admin uses a different component library than previously (Material UI).  

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

## Validation
Code review for this PR should focus only on files that are within the admin app. Other files have already been code reviewed in the previous PR. The same goes for testing in the UI, the admin app is the focus of this PR, the other parts have already been tested. The app should have all the same functionality that it did before the upgrade.

### Links
Link to react-router upgrade PR: https://github.com/transcom/mymove/pull/10020
React-admin upgrade guide: https://marmelab.com/react-admin/Upgrade.html

### CODE REVIEW

File: src/pages/Admin/OfficeUsers/OfficeUserList.jsx [hunk 1]
```diff
@@ -10,26 +10,26 @@ import {
...
-import PropTypes from 'prop-types';
 
 import ImportOfficeUserButton from 'components/Admin/ImportOfficeUserButton';
 import AdminPagination from 'scenes/SystemAdmin/shared/AdminPagination';
 
 // Overriding the default toolbar to add import button
-const ListActions = (props) => {
-  const { total, resource, sort, filterValues, exporter } = props;
+const ListActions = () => {
+  const { total, resource, sort, filterValues, exporter } = useListController();
```
Comment (PDickeyTruss) [inline src/pages/Admin/OfficeUsers/OfficeUserList.jsx:21]: This change fixes [MB-15276](https://dp3.atlassian.net/browse/MB-15276)

**Comment by PDickeyTruss** [comment]
Great catches @duncan-truss! I have pushed updates to address both your comments. Admin forms now have all the same width fields as before the upgrade and office user export includes the roles comma separated (no space). I also fixed up the the admin forms validation that we were noticing wasn't working very well when discussing this earlier. Turns out some of the defaults around form validation changed with the upgrade in an unfavorable way. I was able to add a few props to switch the validation mode back and now validation and related messaging is working as expected.

**Review by duncan-truss — APPROVED** [review]
Admin field changes and import all worked for me :shipit:

**Review by dinabeigelman — APPROVED** [review]
LGTM!

**Pull request merged**